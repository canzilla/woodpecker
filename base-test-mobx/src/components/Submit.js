/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import Button from '@material-ui/core/Button';

export default function Submit(){
    return(
      <div test-attr="test-submit-div">
        <Button test-attr="test-submit-button" variant="contained">Save</Button>
      </div>
    );
}