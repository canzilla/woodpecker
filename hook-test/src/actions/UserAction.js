import {UPDATE_USER, SNACKBAR_UPDATE_STATUS} from "../consts/UserConsts";

export function User_Update(data) {
    return {
        type:UPDATE_USER,
        userObject:data['userObject'],
        userId:data['userId']
    }
}

export function Snackbar_Update_Status(showStatus) {
    return {
        type:SNACKBAR_UPDATE_STATUS,
        showStatus:showStatus
    }
}