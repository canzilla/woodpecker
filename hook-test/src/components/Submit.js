import React from 'react';
import Button from '@material-ui/core/Button';
import {UserContext} from "../context/UserContext";
import {User_Update} from "../actions/UserAction";


export default function Submit(props) {
    const {dispatch} = React.useContext(UserContext);
    const [showSubmit, setShowSubmit] = React.useState(false);
    props.bridgeState.showSubmit = setShowSubmit;

    return (
        <div test-attr="test-submit-div" bridgestate={props.bridgeState}>
            {divContend()}
        </div>
    );

    function divContend() {
        return showSubmit ? createButton() : null
    }

    function createButton() {
        return (
            <Button
                test-attr="test-submit-button"
                variant="contained"
                onClick={() => updateUserObject()}
            >Save</Button>
        );
    }

    function updateUserObject() {
        props.bridgeState['userId'] = props.userId;
        //dispatch(User_Update(props.bridgeState));
        setShowSubmit(false);
    }
}
