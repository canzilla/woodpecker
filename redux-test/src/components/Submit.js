import React, { Component } from 'react';
import { connect } from 'react-redux';
import Button from '@material-ui/core/Button';
import { sendTypedInformationToService } from '../actions/action';

export class Submit extends Component {
    constructor(props){
        super(props);

        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(){
        this.props.sendTypedInformationToService(this.props.objectForDisplay);
    }

    render() {
        const isInformationTyped = this.props.isInformationTyped;
        let buttonComponent = isInformationTyped ?
            (
              <Button 
                element-identifier="submit-button"
                data-testid="submit-button"
                variant="contained"
                onClick={this.handleClick}
              >
                Save
              </Button>
            )
            :
            null;
        return (
            <div element-identifier="component-submit" data-testid="component-submit">
                { buttonComponent }
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        isInformationTyped: state.isInformationTyped,
        objectForDisplay: state.objectForDisplay
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        sendTypedInformationToService: (object) => dispatch(sendTypedInformationToService(object))
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Submit);