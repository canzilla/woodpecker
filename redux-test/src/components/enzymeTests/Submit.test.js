import React from 'react';
import { shallow } from 'enzyme';
import Submit from '../Submit';
import { storeFactory, findComponentByAttribute } from '../../../test/testUtils';
import { isInformationTyped, objectForDisplay } from '../../actions/action';

const setup = (initialState = {}) => {
  const store = storeFactory(initialState);
  return shallow(<Submit store={store} />).dive().dive();
}

describe('render', () => {
  describe('any information has not been type', () => {
    let wrapper;
    beforeEach(() => {
      const initialState = { isInformationTyped: false };
      wrapper = setup(initialState);
    });
    test('render submit component success', () => {
      const component = findComponentByAttribute(wrapper, 'component-submit');
      expect(component.length).toBe(1);
    });
    test('not render submit button', () => {
      const component = findComponentByAttribute(wrapper, 'submit-button');
      expect(component.length).toBe(0);
    });
  });
  describe('any information has been type', () => {
    let wrapper;
    beforeEach(() => {
      const initialState = { isInformationTyped: true };
      wrapper = setup(initialState);
    });
    test('render submit component success', () => {
      const component = findComponentByAttribute(wrapper, 'component-submit');
      expect(component.length).toBe(1);
    });
    test('render submmit button', () => {
      const component = findComponentByAttribute(wrapper, 'submit-button');
      expect(component.length).toBe(1);
    });
  });
});

describe('getting data from global state', () => {
  let store;
  beforeEach(() => {
    store = storeFactory();
  });
  test('getting isInformationTyped', () => {
    store.dispatch(isInformationTyped(true));
    const newState = store.getState();
    const wrapper = setup(newState);
    const isInformationTypedProp = wrapper.instance().props.isInformationTyped;
    expect(isInformationTypedProp).toBe(true);
  });
  test('getting objectForDisplay', () => {
    store.dispatch(objectForDisplay({ name: 'Çağrı' }));
    const newState = store.getState();
    const wrapper = setup(newState);
    const objectForDisplayProp = wrapper.instance().props.objectForDisplay;
    expect(objectForDisplayProp).toStrictEqual({ name: 'Çağrı' });
  });
});

describe('click submit button', () => {
  let wrapper;
  let spy;
  beforeEach(() => {
    const initialState = {
      isInformationTyped: true,
      objectForDisplay: {
        name: 'Çağrı',
        surname: 'Yenice'
      }
    };
    wrapper = setup(initialState);
    spy = jest.spyOn(wrapper.instance(), 'handleClick');
    wrapper.instance().forceUpdate();
  });
  test('not click', () => {
    expect(spy).not.toHaveBeenCalled();
  });
  test('click', () => {
    const component = findComponentByAttribute(wrapper, 'submit-button');
    component.simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});