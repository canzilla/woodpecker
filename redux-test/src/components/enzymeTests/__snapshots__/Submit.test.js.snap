// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`click submit button snapshot testi 1`] = `
<div
  data-testid="component-submit"
  element-identifier="component-submit"
/>
`;
